refinery/core/models.py

```diff
@@ -801,7 +790,7 @@ def _dataset_delete(sender, instance, *args, **kwargs):
     with transaction.atomic():
         # delete FileStoreItem and datafile corresponding to the
         # metadata file used to generate the DataSet
-        instance.get_metadata_as_file_store_item().delete()
+        instance.get_investigation().get_file_store_item().delete()
```

Comment:
Shouldn't this `FileStoreItem` be deleted automatically on deletion of `Investigation`?